pallets/file-system/src/utils.rs

```diff
@@ -1944,13 +1958,24 @@ where
         // Decrease data used by the SP.
         <T::Providers as MutateStorageProvidersInterface>::decrease_capacity_used(&sp_id, size)?;
 
-        // If the new capacity used is 0 and the Provider is a BSP, stop its randomness cycle.
         if <T::Providers as ReadStorageProvidersInterface>::is_bsp(&sp_id)
-            && <T::Providers as ReadStorageProvidersInterface>::get_used_capacity(&sp_id)
-                == Zero::zero()
+            && new_root == <T::Providers as shp_traits::ReadProvidersInterface>::get_default_root()
         {
+            // If it doesn't store any files we stop the challenge cycle and stop its randomness cycle.
+
+            let used_capacity =
+                <T::Providers as ReadStorageProvidersInterface>::get_used_capacity(&sp_id);
+            if used_capacity == Zero::zero() {
+                // Emit event if we have inconsistency. We can later monitor for those.
```

Comment:
Wouldn't it be the other way around? The inconsistency would be if the capacity IS NOT zero.

```suggestion
            if used_capacity != Zero::zero() {
                // Emit event if we have inconsistency. We can later monitor for those.
```